src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc

```diff
@@ -98,26 +98,39 @@ void VehicleInfoPlugin::OnApplicationEvent(
 }
 
 void VehicleInfoPlugin::UnsubscribeFromNonExistingVDItems() {
-  auto applications = application_manager_->applications();
-  auto& std_vehicle_data = application_manager::MessageHelper::vehicle_data();
-
-  for (auto& app : applications.GetData()) {
-    auto& ext = VehicleInfoAppExtension::ExtractVIExtension(*app);
-    auto subscription_names = ext.Subscriptions();
-    for (auto& subscription_name : subscription_names) {
-      const bool is_std_vd_name =
-          std_vehicle_data.end() != std_vehicle_data.find(subscription_name);
-      const bool is_valid_custom_vd_name =
-          custom_vehicle_data_manager_->IsValidCustomVehicleDataName(
-              subscription_name);
-      if (!is_std_vd_name && !is_valid_custom_vd_name) {
-        LOG4CXX_DEBUG(logger_,
-                      "Vehicle data item " << subscription_name
-                                           << " has been revoked by policy");
-        ext.unsubscribeFromVehicleInfo(subscription_name);
+  typedef std::vector<std::string> StringsVector;
+
+  auto get_items_to_unsubscribe = [this](StringsVector& out_items) {
+    auto applications = application_manager_->applications();
+    for (auto& app : applications.GetData()) {
+      auto& ext = VehicleInfoAppExtension::ExtractVIExtension(*app);
+      auto subscription_names = ext.Subscriptions();
+      for (auto& subscription_name : subscription_names) {
+        if (custom_vehicle_data_manager_->IsRemovedCustomVehicleDataName(
+                subscription_name)) {
+          ext.unsubscribeFromVehicleInfo(subscription_name);
+          if (!helpers::in_range(out_items, subscription_name)) {
+            LOG4CXX_DEBUG(logger_,
+                          "Vehicle data item "
+                              << subscription_name
+                              << " has been removed by policy");
+            out_items.push_back(subscription_name);
+          }
+        }
       }
     }
+  };
+
+  StringsVector items_to_unsubscribe;
+  get_items_to_unsubscribe(items_to_unsubscribe);
```

Comment:
@AKalinich-Luxoft I would rather use it like 
```cpp
StringsVector items_to_unsubscribe =get_items_to_unsubscribe();
```

Reply:
@AByzhynar I choose the option 1 with get_ prefix. It's a labmda and I am considering it as a local variable so kept appropriate naming. Please take a look at https://github.com/smartdevicelink/sdl_core/pull/3081/commits/7aa5625d93c526d8a1332fa5ae3c32e0c4a6a02c